services/dynamodb/src/main/java/software/amazon/awssdk/services/dynamodb/utils/AttributeValueHelper.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.dynamodb.utils;
+
+import java.nio.ByteBuffer;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+public final class AttributeValueHelper {
+    private AttributeValueHelper() {}
+
+    public static AttributeValue attr(String s) {
+        return AttributeValue.builder().s(s).build();
+    }
+
+    public static AttributeValue attr(Number n) {
+        return AttributeValue.builder().n(String.valueOf(n)).build();
+    }
+
+    public static AttributeValue attr(Boolean bool) {
+        return AttributeValue.builder().bool(bool).build();
+    }
+
+    public static AttributeValue attr(ByteBuffer binary) {
+        return AttributeValue.builder().b(binary).build();
+    }
+
+    public static AttributeValue attr(byte[] binary) {
+        return attr(ByteBuffer.wrap(binary));
+    }
+
+    public static AttributeValue attr(Object object) {
```

Comment:
What's this used for? Can we have a use method name so that we can have type checking on the method parameters? `AttributeValueHelper.attr(new Object())` would be nice to catch at compile time. A reasonable person could assume that if the method takes an object, it would work.

Reply:
This is used to convert an unknown but supported object to the appropriate `AttributeValue` representation - fair about the assumption that I could pass object - I'll document.

Regarding the type-checked methods, that's what all the above `attr` methods are - specifically typed methods.

Reply:
Sorry, I did a bad job voicing the complaint.

They're all overloads, so if I was wondering "does `AttributeValue.attribute` support my type?" I'd look through the overloads and see "Oh cool, there's one for object, I can use anything" and then use it for my Map, List, Banana, etc.

If it was a different method name that makes it clear that "Yeah, we said Object, but it's not actually all types" I probably wouldn't have made that mistake... `AttributeValue.unsafeAttribute(Object)`?